nil/internal/execution/state.go

```diff
@@ -1130,6 +1148,14 @@ func (es *ExecutionState) HandleTransaction(
 		}()
 	}
 
+	if txn.IsInternal() {
```

Comment:
Could we move this check to `ValidateInternalMessage`?

Then we can do `es.InTxCounts[txn.From.ShardId()] = nextTx + 1` in defer.